pandas/io/msgpack/_unpacker.pyx

```diff
@@ -129,8 +127,14 @@ def unpackb(object packed, object object_hook=None, object list_hook=None,
         Py_ssize_t buf_len
         char* cenc = NULL
         char* cerr = NULL
-
-    PyObject_AsReadBuffer(packed, <const void**>&buf, &buf_len)
+        Py_buffer view
+
+    # Effectively re-implement deprecated PyObject_AsReadBuffer;
+    # based on https://xpra.org/trac/ticket/1884
+    PyObject_GetBuffer(packed, &view, PyBUF_SIMPLE)
+    buf = <char*>view.buf
+    buf_len = view.len
+    PyBuffer_Release(&view)
```

Comment:
Shouldn't this release call come later in the function, after `buf` and `buf_len` are used?

Reply:
Yes. There is no guarantee in the buffer protocol that the buffer stays alive beyond a call to `PyBuffer_Release()`.
Also, `PyObject_GetBuffer()` can fail, in which case it will return `-1`.

Simply casting `packed` to a `<char*>` might actually also work in Cython, but that won't give you the buffer length.  Casting it to a memory view would, but that's also a bit more heavy than calling `GetBuffer` directly.

Reply:
@jbrockmendel can you move the release to the end of the function? (for both the success and unsuccessful paths)